paasta_tools/mesos_maintenance.py

```diff
@@ -334,6 +379,118 @@ def load_credentials(mesos_secrets='/nail/etc/mesos-slave-secret'):
     return username, password
 
 
+def get_principal(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the principal from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the principal/username
+    """
+    return load_credentials(mesos_secrets)[0]
+
+
+def get_secret(mesos_secrets='/nail/etc/mesos-slave-secret'):
+    """Helper function to get the secret from the mesos-slave credentials
+    :param mesos_secrets: optional argument specifying the path to the file containing the mesos-slave credentials
+    :returns: a string containing the secret/password
+    """
+    return load_credentials(mesos_secrets)[1]
+
+
+def reserve(slave_id, resource, amount):
+    """Dynamically reserve resources in marathon to prevent tasks from using them.
+    :param slave_id: the id of the mesos slave
+    :param resource: the mesos resource type we want to reserve (i.e. cpus, mem, gpus, etc)
+    :param amount: integer representing the amount of the above resource to reserve
+    :returns: boolean where 0 represents success and 1 is a failure
+    """
+    log.info("Dynamically reserving %f %s on %s" % (amount, resource, slave_id))
+    payload = {
+        'slaveId': slave_id,
+        'resources': str(build_reservation_payload(slave_id, resource, amount)).replace("'", '"').replace('+', '%20')
```

Comment:
Do we have to do the string replacing? I guess mesos is expecting JSON in a application/x-www-form-urlencoded POST which admittedly seems pretty nasty of it.

What about `json.dumps` and `urllib.quote`. Also shouldn't `+` be `%2B`?